src/react/img-file.ts

```diff
@@ -47,17 +72,40 @@ export function ImgFile({ file, meta, ...imgProps }: ImgFileProps) {
           break;
       }
 
+      // Clean up previous object URL if it exists and we're loading a new file
+      if (fileObjRef.current !== fileObj && cleanupRef.current) {
+        cleanupRef.current();
+        cleanupRef.current = null;
+      }
+
       if (fileObj && /image/.test(fileType)) {
-        // eslint-disable-next-line no-restricted-globals
-        const src = URL.createObjectURL(fileObj);
-        setImgDataUrl(src);
-        // eslint-disable-next-line no-restricted-globals
-        return () => URL.revokeObjectURL(src);
+        // Skip if it's the same exact file object
+        if (fileObjRef.current !== fileObj) {
+          const src = getObjectUrl(fileObj);
+          setImgDataUrl(src);
+          fileObjRef.current = fileObj;
+
+          // Store cleanup function
+          cleanupRef.current = () => {
+            const cacheKey = getCacheKey(fileObj);
+            if (objectUrlCache.has(cacheKey)) {
+              // eslint-disable-next-line no-restricted-globals
+              URL.revokeObjectURL(objectUrlCache.get(cacheKey) as string);
+              objectUrlCache.delete(cacheKey);
+            }
+          };
+
+          return cleanupRef.current;
+        }
+
+        // Return existing cleanup if same file
+        return cleanupRef.current;
       }
+      return null;
     };
 
     let isMounted = true;
-    let cleanup: (() => void) | undefined;
+    let cleanup: (() => void) | null = null;
```

Comment:
ok react is more null than undefined but --- the all feels pretty hacky. 
sorry I should provide a better solution -:(